### PR TITLE
updated branch with engagements owning users 

### PR DESCRIPTION
**Pull request by julies642**
Have Engagements own actual users rather than just student strings. Updated views appropriately. Had to make a new branch with changes from eng_users branch which is not properly updated.

### CODE REVIEW

File: app/views/apps/show.html.haml [hunk 1]
```diff
@@ -63,7 +63,7 @@
         %td= mail_to(engagement.contact.email, engagement.contact.name) if engagement.contact
         %td= [engagement.coach.try(:name),engagement.coaching_org.try(:name)].join(', ')
         %td= engagement.team_number
-        %td= engagement.student_names
+        %td <div class="members"><div class="member1">#{engagement.users[0].try(:name)} </div><div class="member2">#{engagement.users[1].try(:name)}</div><div class="member3">#{engagement.users[2].try(:name)}</div><div class="member4">#{engagement.users[3].try(:name)}</div><div class="member5">#{engagement.users[4].try(:name)}</div><div class="member6">#{engagement.users[5].try(:name)}</div></div>
```
Comment (kimhoang73) [inline app/views/apps/show.html.haml]: can we do some sort of loop here instead?

**Review by kimhoang73 — APPROVED** [review]
i think it would be nice if we could make the show view have a loop to go thru the users if possible! but everything else looks good!

File: app/views/apps/show.html.haml (hunk 1 above)
Comment (csungwon) [inline app/views/apps/show.html.haml]: I agree with Kim. I think there should not be a constraint on the number of developers an engagement can have.

Comment (kyriakipapadimitriou) [reply app/views/apps/show.html.haml]: The loop implementation is definitely the way to go but Julie and I have currently some issues displaying the names under the correct column "Developers". When we use the loop, names are displayed on top of the table, namely before "Selected". We will fix this during the week after consulting with Andrew.

Comment (kyriakipapadimitriou) [reply app/views/apps/show.html.haml]: The loop implementation is definitely the way to go but Julie and I have currently some issues displaying the names under the correct column "Developers". When we use the loop, names are displayed on top of the table, namely before "Selected". We will fix this during the week after consulting with Andrew.

File: app/views/apps/show.html.haml [hunk 2]
```diff
@@ -63,7 +63,10 @@
         %td= mail_to(engagement.contact.email, engagement.contact.name) if engagement.contact
         %td= [engagement.coach.try(:name),engagement.coaching_org.try(:name)].join(', ')
         %td= engagement.team_number
-        %td= engagement.student_names
+        %td
+          %div{:class => 'members'}
```
Comment (csungwon) [inline app/views/apps/show.html.haml:67]: This line could be removed.

**Review by csungwon — APPROVED** [review]
I think overall it looks good. One thing you want to work on is to delete the student names in the engagement/edit page. You can delete the student names attribute from the engagement table, but let's keep the attribute for now, because we might need the information in order to add them to users list. Just get rid of the student names form field.

File: app/views/apps/show.html.haml (hunk 2 above)
Comment (csungwon) [reply app/views/apps/show.html.haml:67]: ah it was just an indentation issue. I will fix this.

**Pull request merged**